co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/ambassador/CarlaAmbassador.java

```diff
@@ -351,15 +357,29 @@ public synchronized void processTimeAdvanceGrant(long time) throws InternalFeder
                 nextTimeStep += carlaConfig.updateInterval * TIME.MILLI_SECOND;
                 isSimulationStep = false;
             }
+            // TODO: What is this. Why are we request a time advance based on this counter and 
+            // what is it counting. It is labeled as counting the times we attempt to connect to 
+            // CARLA but it seems to increment every time processTimeAdvanceGrant is called
             rti.requestAdvanceTime(nextTimeStep + this.executedTimes, 0, (byte) 2);
```

Comment:
@kjrush Do you have any information why were are requesting time advance like this with a continuously incremented counter? If you have any information about executedTimes and why its used in our requestAdvanceTime.

Reply:
I did a quick skim of the rest of this PR, but so far focused on this.

I don't really see what purpose this executed times has, to be honest. It might just be entirely unnecessary. For the record, it doesn't increment infinitely. It's actually reset later on in the code inside the `receiveInteraction(SimulationStepResponse)` method. The interactions being received there seem to be generated whenever SUMO provides a new TraCI update, so it should be frequently, so this should never go above a small value, maybe 10 at most.

But even 10 doesn't make much sense, considering this `requestAdvanceTime` arg is in milliseconds, and there's no multiplicative factor applied to this `executedTimes` value (unlike `carlaConfig.updateInterval` above), so it's small enough it probably doesn't actually do anything. I imagine it was implemented like this with the goal that if we don't have any fresh data from sumo, we should wait longer to delay the CARLA updates until we do have fresh data. But I'm not sure that's needed and it seems unlikely this is actually accomplishing that.

I think you were right to be questioning about this line, but in the end I'm not sure it matters one way or the other. I'd be in favor of just removing it, but keeping this line in our mental bookmarks in case we run into issues with the ambassador. If we hit those issues we'll have this top of mind to re-investigate.